Yank/utils.py

```diff
@@ -1731,6 +1732,75 @@ def _sanitize_unit_name(unit_name):
         return unit_name
 
 
+def generate_development_feature(feature_dict):
+    """
+    Helper function for generating a class which can flag classes, tests, and functions that are developmental.
+
+    Output class not quite a mixin because it has to be the first class due to the `__init__` flag
+
+    Parameters
+    ----------
+    feature_dict : dict
+        Dictionary of form "test_string : pre-computed test" where "test_string" is just an identifier and
+        "pre-computed test" is a boolean-like object, usually the result of some test. All pre-computed tests will
+        be cast to bool
+
+    Returns
+    -------
+    DevelopmentFeature : class
+        Class which checks against the feature_dict and can be used in several ways:
+
+        * Class Inherited: When inherited as a class, calling its ``__init__()`` will raise an error if features are
+          not met
+        * True/False check function: When calling ``dev_validate()`` will return bool if all features are true.
+        * True/False decorator: When decorating function with ``dev_validation``, function will only be called if
+          ``dev_validate()`` would return True, otherwise simply returns. Helpful for running tests.
+        * Dict of reasons: Property ``dev_reasons`` will return the dictionary of failed dependencies
+        * Dict of all: Property ``dev_features`` will return the dictionary of features it expects and their tests
+
+        With the exception of the `__init__``, all other functions are properties are Class based and do not
+        require instantiation. Function names are all given the `dev_` prefix to avoid clashes with other names its
+        a part of its psudo-mixin properties
+    """
+    base_err = ('This feature cannot be used because it has been marked as "Developmental" and ' 
+                'the following conditions have not been met:\n')
+    valid = True  # Assume valid until proven otherwise
+    check_dict = {}
+    for test_string, test in feature_dict.items():
+        test = bool(test)  # Cast tests to bool
+        if not test:
+            base_err += "\t- {}: {}\n".format(test_string, test)  # Add to error message
+            valid = False  # Not all features met
+            check_dict = {**check_dict, **{test_string: test}}  # Create check sub-dict
+
+    class DevelopmentFeature(object):
+        DEV_ERROR = base_err if not valid else None
+        dev_reasons = check_dict
+        dev_features = feature_dict
+        dev_validate = valid
+
+        def __init__(self, *args, **kwargs):
+            if not self.dev_validate:
```

Comment:
I think you'll have to call `super().__init__(*args, **kwargs)` here or this will likely have problems when used as a mix-in. I think this is working right now because `ReceptorLigandRestraint` doesn't implement a constructor.

Reply:
It will, which is why its only a "psuedo"-mixin.

We can fix it in a later PR